figma-shop/src/app/shared/services/filters.service.ts

```diff
@@ -0,0 +1,67 @@
+import { Injectable } from '@angular/core';
+import { ProductInterface } from '../interfaces/product.interface';
+import { FilterData } from '../interfaces/filter-data.interface';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class FiltersService {
+  public filter: FilterData;
+
+  constructor() { }
+
+  getFilterData(data): void {
+    this.filter = data;
+  }
+
+  filterAllProducts(products: ProductInterface[]): ProductInterface[] {
+    if (!this.filter || this.filter.category === null && this.filter.brand.length === 0 &&
+      this.filter.rating.length === 0) {
```

Comment:
if (!this.filter || this.filter.category === null && this.filter.brand.length &&
      this.filter.rating.length)

Reply:
thanks, looks much better